libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3632,6 +3632,27 @@ void TypeChecker::endVisit(Literal const& _literal)
 
 void TypeChecker::endVisit(UsingForDirective const& _usingFor)
 {
+	if (_usingFor.global())
+	{
+		if (m_currentContract)
+			solAssert(m_errorReporter.hasErrors());
+		if (Declaration const* typeDefinition = _usingFor.typeName()->annotation().type->typeDefinition())
+		{
+			if (typeDefinition->scope() != m_currentSourceUnit)
+				m_errorReporter.typeError(
+					4117_error,
+					_usingFor.location(),
+					"Can only use \"global\" with types defined in the same source unit at file level."
+				);
+		}
+		else
+			m_errorReporter.typeError(
+				8841_error,
+				_usingFor.location(),
+				"Can only use \"global\" with user-defined types."
+			);
+	}
+
```

Comment:
Something must be still uninitialized at this point because [soltest is crashing due to a memory violation](https://app.circleci.com/pipelines/github/ethereum/solidity/23256/workflows/076c71e3-cd11-4dce-9d0a-63012cb529bf/jobs/1021108):
```
memory access violation at address: 0x00000000: no mapping at fault address
```
And it's this test that crashes it:
```solidity
using {f} for * global;
function f(uint) pure{}
```

There are also crashes in the fuzzer during CLI tests (e.g. [1021105](https://app.circleci.com/pipelines/github/ethereum/solidity/23256/workflows/076c71e3-cd11-4dce-9d0a-63012cb529bf/jobs/1021105)) but they could just be caused by the same memory violation.